mdn/developer-portal
# Support HTTPS for local development.

This changeset addresses a need for serving the local build using HTTPS.

It adds nginx + SSL cert generation into the local docker stack, following [how we do it for kuma](https://github.com/mdn/kuma/blob/master/docker-compose.ssl.yml)

Note that SSL is now the default for the local stack, but http://localhost:8000 still responds.  
Ideally all local development should be done on https://developer-portal-127-0-0-1.nip.io so that future changes (eg CSP support) don't break belocal behaviour. If you want to work offline, you can add `127.0.0.1    developer-portal-127-0-0-1.nip.io` to your `/etc/hosts`

Local fixtures are updated so that the default Wagtail site is expecting to be on `developer-portal-127-0-0-1.nip.io`, and this is matched by a new default for `BASE_URL`, so that the static build still works.

Tried a static build and it still works fine.

Nothing in this changeset should affect non-local behaviour.

(Resolves #424)

## How to test

- `docker-compose down` of your local build, if running
- check out branch
- `docker-compose build && docker-compose up`
- visit the site at `https://developer-portal-127-0-0-1.nip.io`
- visit the CMS at `https://developer-portal-127-0-0-1.nip.io/admin/` - you'll need to re-authenticate because the cookie domain is different
- visit both at `http://localhost:8000` to see you can still get to it the old way (Wagtail should work exactly the same, but the rendered site may break once we add CSP rules etc)
